lib/Models/RegionProvider.js

```diff
@@ -0,0 +1,405 @@
+        if (results.regionValues[index] !== undefined && !dataset.hasTimeData()) {
+            // if there is a time column, ignore ambiguous matches
+            results.ambiguousMatches[codes[row]] = true;
+            continue;
+        }
+        if (defined(vals)) {
+            results.regionValues[index] = vals[row];
+        } else {
+            results.regionValues[index] = 0; // In some cases region mapping without a data variable is meaningful.
+        }
+        if (defined(regionVar)) {
+            // set reverse lookup from variable back to region index.
+            regionVar.setRegionCode(i, this.regions[index].id);
+        }
+        results.successes ++;
+    }
+    return results;
+};
+
+/**
+ * Pre-generates a function which quickly turns a value into a colour.
+ * @param {Number[]} regionValues Array of values.
+ * @param {Function} colorFunc should be function(val) { return [r,g,b,a]; } 
+ * @returns {Function} Function of type f(regionIndex) { return [r,g,b,a]; }
+ */
+RegionProvider.prototype.getColorLookupFunc = function(regionValues, colorFunc) {
+    var colors = regionValues.map(colorFunc);
+    return function(regionIndex) {
+        // actually maybe it's ok to return undefined after all
+        return colors[regionIndex] !== undefined ? colors[regionIndex] : [0,0,0,0]; //# is it important to get rid of this if?
+    };
+
+};
+
+
+
+/*
+ * @returns The index of the first column that left-matches any of the given aliases 
+ * @param varNAmes Array of variable names.
+ */
+RegionProvider.prototype.findRegionVariable = function(varNames) {
+    for (var j = 0; j < this.aliases.length; j++) {
+        for (var i = 0; i < varNames.length; i++) {
+            //check that it is the right type of code
+            //if (typeof this.regionTextCode === "boolean" && this.regionTextCode !== variables[i].isTextCode) {
+            //    continue;
+            //}
+            // SB: I don't think that's a worthwhile test
+
+            //if (varNames[i].toLowerCase() === this.aliases[j]) {
+            if (new RegExp('^' + this.aliases[j] + '$', 'i').test(varNames[i])) {
+                return i; 
+            }
+        }
+    }
+    return -1;
+};
+
+RegionProvider.prototype._getDisambigProvider = function() {
+    if (!defined(this.disambigColumns) || this.disambigColumns.length < 1 || !defined(this.regionProviderList)) {
+        return undefined;
+    }
+    var disambig = this.disambigColumns[0];
+    return this.regionProviderList.getRegionProvider(disambig.region);
+};
+
+/**
+ * If a disambiguation column is known for this provider, return a column matching its description.
+ */
+RegionProvider.prototype.findDisambigVariable = function(variables) {
+    var dabp = this._getDisambigProvider();
+    if (!defined(dabp)) {
+        return undefined;
+    }
+    var dv = variables[dabp.findRegionVariable(variables)];
+    return (defined(dv) ? dv.name : undefined);
+};
+// this probably doesn't belong here
```

Comment:
yeah, I agree.  It's ok for `RegionProvider` to offer up this information in an object literal or something like that, but it shouldn't be formatting it as HTML.